tests/unit/db/feed_tests.py

```diff
@@ -0,0 +1,544 @@
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertTrue(str(feed.last_seq).startswith('9'))
+
+    def test_get_feed_using_style_all_docs(self):
+        """
+        Test getting content back for a feed using style set to "all_docs"
+        """
+        self.populate_db_with_documents(3)
+        for i in range(3):
+            docid = 'julia{0:03d}'.format(i)
+            doc = self.db[docid]
+            doc.delete()
+            with Document(self.db, docid) as doc:
+                doc['name'] = 'Jules'
+                doc['age'] = i
+        feed = Feed(self.session, self.changes_url, style='all_docs')
+        changes = list()
+        for change in feed:
+            self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+            if not self.cloudant_test:
+                self.assertEqual(len(change['changes']), 2)
+            changes.append(change)
+        expected = set(['julia000', 'julia001', 'julia002'])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertTrue(str(feed.last_seq).startswith('9'))
+
+    def test_get_feed_using_since(self):
+        """
+        Test getting content back for a feed using the since option
+        """
+        self.populate_db_with_documents(3)
+        feed = Feed(self.session, self.changes_url)
+        changes = [change for change in feed]
+        last_seq = feed.last_seq
+        self.populate_db_with_documents(3, off_set=3)
+        feed = Feed(self.session, self.changes_url, since=last_seq)
+        changes = list()
+        for change in feed:
+            self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+            changes.append(change)
+        expected = set(['julia003', 'julia004', 'julia005'])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertTrue(str(feed.last_seq).startswith('6'))
+
+    def test_get_feed_using_since_now(self):
+        """
+        Test getting content back for a feed using since set to "now"
+        """
+        self.populate_db_with_documents(3)
+        feed = Feed(self.session, self.changes_url,
+            feed='continuous', heartbeat=1000, since='now')
+        changes = list()
+        first_pass = True
+        beats = 0
+        for change in feed:
+            if first_pass and not change:
+                self.populate_db_with_documents(3, off_set=3)
+                first_pass = False
+                continue
+            elif change:
+                self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+                changes.append(change)
+                beats = 0
+            else:
+                beats += 1
+            if beats == 15 or len(changes) == 3:
+                feed.stop()
+        expected = set(['julia003', 'julia004', 'julia005'])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+
+    def test_get_feed_using_timeout(self):
```

Comment:
What about another test case to assert that ``heartbeat`` supersedes ``timeout``?